tests/trino/test-values.yaml

```diff
@@ -15,6 +15,23 @@ server:
     query.execution-policy=phased
   autoscaling:
     enabled: true
+  keda:
+    enabled: true
```

Comment:
If we have two autoscaling mechanism, we should test Keda in a separate test.

Reply:
I would like to point out that the existing HPA (enabled by `server.autoscaling.enabled`) requires Metrics Server, or some other service, backing the Kubernetes Metrics API that provides CPU and memory usage to scale workers up or down. Since no such service is installed during the tests, setting `server.autoscaling.enabled=true` has no real effect other than creating the HorizontalPodAutoscaler resource. Workers are not really scaling up or down.

Should we install the Metrics Server to actually test HPA? 

We would need to set resource requests so that HPA can figure out what the percentage thresholds mean (like if memory request is 2Gi, and usage is 1Gi then in percentage it is 50%, without the request HPA cannot work).  

Also, we would need queries that end up using a certain amount of CPU/memory so that they can trigger the HPA to scale up or down (maybe scaling down is enough for the test, like we could start with x workers and then wait for them to scale down to 1 with HPA).

The nice thing about KEDA is that you can start with no workers, and you implicitly know that scaling worked just because you got the query response (the query could not have run without workers).

Final consideration, KEDA does not need the Metrics Server to work, it can just use Prometheus. Installing Prometheus and KEDA takes some time, so using a separate test will take some time to run. 

Anyway, I am open to suggestions :)

Reply:
I naively hoped it'll be easy to have a similar test with HPA. Let's leave it as is and we can improve tests later. I was planning on using pytest for managing fixtures, so we can see if it'll make it easier to rearrange tests.

Reply:
Out of curiosity, have you considered using [testcontainers](https://pypi.org/project/testcontainers/) to spin up kind and use it as a fixture in pytest?

Reply:
Yes, testcontainers are great!